tesla_powerwall/api.py

```diff
@@ -245,3 +246,7 @@ def get_site_info_grid_codes(self) -> list:
 
     def post_site_info_site_name(self, body: dict) -> dict:
         return self.post("site_info/site_name", body)
+
+    def post_islanding_mode(self, body: dict) -> dict:
+        payload = json.dumps(body)
```

Comment:
For some reason this was the missing link / bug breaking my code.

If there is a better way to do this, please let me know!

Reply:
oh yeah, that is probably my fault. You can remove that json dumps line here and rather change the `API._post` method, by replacing the `data` parameter with the `json` param. Then the content will automatically be serialized and the content type will also be correct. For reference the last section in https://requests.readthedocs.io/en/latest/user/quickstart/#more-complicated-post-requests

Reply:
I've updated this (or something like it, might have misunderstood sorry!).

Integration tests are still passing, though there is one method in particular (`post_sitemaster_run_for_commissioning`) that doesn't seem to have an integration test so may be effected.

Let me know if there's anything else this needs! :)